### PR TITLE
[BugFix] Bind storage volume to table when it has been created successfully

### LINKED ISSUE #26160: [storage volume] failed drop storage volume
### Steps to reproduce the behavior (Required)
```
CREATE STORAGE VOLUME
mysv
TYPE = S3
LOCATIONS = ("s3://xxx/xxxxx/")
COMMENT "default storage"
PROPERTIES(
    "aws.s3.region" = "xxxxxxxxx",
    "aws.s3.endpoint" = "https://xxxxxxxxxxx.com",
    "aws.s3.use_aws_sdk_default_behavior" = "false",
    "aws.s3.use_instance_profile" = "false",
    "aws.s3.access_key" = "xxxxxxxxxxx",
    "aws.s3.secret_key" = "xxxxxxxxxx"
);
```
create table failed:
```
create database db1 properties ("storage_volume"="mysv");
use db1;
CREATE TABLE t1 (
a int,
b int)
duplicate KEY (a)
DISTRIBUTED BY HASH(a);
```
`drop database db1;`
`drop STORAGE VOLUME mysv;`

### Expected behavior (Required)
`Query OK`
### Real behavior (Required)
`ERROR 1064 (HY000): Unexpected exception: Storage volume 'mysv3' is referenced by dbs or tables, dbs: [], tables: [11123]`
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
`branch-3.1-eeed529`

### PR DESCRIPTION
**Pull request by TszKitLo40**
Fixes #26160 

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 5 / 5 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/OlapTableFactory.java|5|5|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/server/OlapTableFactory.java [hunk 1]
```diff
@@ -555,6 +555,11 @@ public Table createTable(LocalMetastore metastore, Database db, CreateTableStmt
         // process lake table colocation properties, after partition and tablet creation
         colocateTableIndex.addTableToGroup(db, table, colocateGroup, true /* expectLakeTable */);
 
+        if (RunMode.getCurrentRunMode() == RunMode.SHARED_DATA && !storageVolumeId.isEmpty()) {
+            StorageVolumeMgr svm = GlobalStateMgr.getCurrentState().getStorageVolumeMgr();
+            svm.bindTableToStorageVolume(storageVolumeId, tableId);
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/server/OlapTableFactory.java:560]: can not bind before write edit log.
or clean up when failed.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 8 / 8 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/OlapTableFactory.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateTableStmt.java|3|3|100.00%|[]|

**Review by wyb — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 8 / 8 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/OlapTableFactory.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateTableStmt.java|3|3|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/server/OlapTableFactory.java [hunk 2]
```diff
@@ -232,329 +232,336 @@ public Table createTable(LocalMetastore metastore, Database db, CreateTableStmt
             throw new DdlException("Unrecognized engine \"" + stmt.getEngineName() + "\"");
         }
 
-        table.setComment(stmt.getComment());
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/server/OlapTableFactory.java:235]: should check `bindTableToStorageVolume` return value

File: fe/fe-core/src/main/java/com/starrocks/server/OlapTableFactory.java [hunk 3]
```diff
@@ -232,329 +232,336 @@ public Table createTable(LocalMetastore metastore, Database db, CreateTableStmt
...
-                Map<String, String> binlogAvailableVersion = table.buildBinlogAvailableVersion();
-                table.setBinlogAvailableVersion(binlogAvailableVersion);
-                LOG.info("set binlog available version when create table, tableName : {}, partitions : {}",
-                        tableName, binlogAvailableVersion.toString());
+
+            // process lake table colocation properties, after partition and tablet creation
+            colocateTableIndex.addTableToGroup(db, table, colocateGroup, true /* expectLakeTable */);
+        } catch (DdlException e) {
+            if (RunMode.getCurrentRunMode() == RunMode.SHARED_DATA) {
+                GlobalStateMgr.getCurrentState().getStorageVolumeMgr().unbindTableToStorageVolume(tableId);
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/server/OlapTableFactory.java:560]: there are some other exceptions in `onCreate` after this.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 163 / 194 (84.02%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/OlapTableFactory.java|155|185|83.78%|[226, 251, 262, 263, 275, 276, 288, 308, 309, 345, 347, 348, 369, 370, 390, 391, 407, 408, 429, 430, 472, 473, 515, 516, 517, 518, 529, 530, 546, 562]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|5|6|83.33%|[809]|
|:large_blue_circle:|com/starrocks/server/StorageVolumeMgr.java|3|3|100.00%|[]|

**Review by wyb — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**